src/ds/ds6/ds6-safety.cpp

```diff
@@ -184,4 +195,57 @@ namespace librealsense
 
         return results;
     }
+
+    void ds6_safety_sensor::set_safety_preset(int index, const rs2_safety_preset& sp) const
+    {
+        //// convert sp to vector of bytes
+        auto data_ptr = const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(&sp));
```

Comment:
Why const_cast?
This is RO data right?

Reply:
done.
optimized the whole function. take a look on the new one